### PR TITLE
First attempt to implement HMAC-based key derivation function

### LINKED ISSUE #477: HKDF support
Hi, do you support HKDF? Like an extension for HMAC?

If yes, how can I use it? And if not, do you have any plan to support it?

Thanks

### PR DESCRIPTION
**Pull request by Interfere**
Fixes #477 
I have noticed, there is a request from community for HKDF.

### CODE REVIEW

File: Sources/CryptoSwift/HKDF.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ///   - keyLength: intended length of derived key
+    public init(password: Array<UInt8>, salt: Array<UInt8>? = nil, info: Array<UInt8>? = nil, keyLength: Int? = nil /* dkLen */, variant: HMAC.Variant = .sha256) throws {
+        guard !password.isEmpty else {
+            throw Error.invalidInput
+        }
+        
+        let dkLen = keyLength ?? variant.digestLength
+        let keyLengthFinal = Double(dkLen)
+        let hLen = Double(variant.digestLength)
+        let numBlocks = Int(ceil(Double(keyLengthFinal) / hLen)) // l = ceil(keyLength / hLen)
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/HKDF.swift]: `keyLengthFinal` is already `Double`

File: Sources/CryptoSwift/HKDF.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    
+    public func calculate() throws -> Array<UInt8> {
+        let hmac = HMAC(key: self.prk, variant: self.variant)
+        var ret = Array<UInt8>()
+        ret.reserveCapacity(self.numBlocks * self.variant.digestLength)
+        var value = Array<UInt8>()
+        for i in 1...self.numBlocks {
+            value.append(contentsOf: self.info)
+            value.append(UInt8(i))
+            value = try hmac.authenticate(value)
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/HKDF.swift]: Instead of overwrite `value`, the new variable would be more readable

**Comment by krzyzanowskim** [comment]
@Interfere I don't know what happened, but the changes are missing in the repo. Either I screw up and force push or something. Would you mind prepare this PR again, please?

**Comment by Interfere** [comment]
I see the commit is in the master branch already.

**Comment by krzyzanowskim** [comment]
yea, sorry for the confusion, I messed up locally. It's there. thanks!

**Pull request merged**